### PR TITLE
chore: replace ioutil with io and os

### PR DESCRIPTION
**Pull request by estensen**
`ioutil` was depcrecated in [Go 1.16](https://go.dev/doc/go1.16)

### CODE REVIEW

**Comment by guseggert** [comment]
What is your motivation for this? 

> The io/ioutil package remains and will continue to work as before

**Comment by estensen** [comment]
`ioutil` will continue to work as before yes, but not improved. If you are walking a directory tree then `os.ReadDir` is more efficient than `ioutil.ReadDir` because of less system calls.

You can see the discussion on the proposal [here](https://github.com/golang/go/issues/41188)

File: coverage/main/main.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
+//go:build testrunmain
```
Comment (Jorropo) [inline coverage/main/main.go:1]: This is out of scope (altho arguably also need to be done).

If you want to make an other PR which just `go fmt ./...` with go1.18 I'll review it.

File: fuse/ipns/ipns_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build !nofuse && !openbsd && !netbsd && !plan9
```
Comment (Jorropo) [inline fuse/ipns/ipns_test.go:1]: Same here

File: fuse/node/mount_test.go [hunk 3]
```diff
@@ -1,9 +1,9 @@
+//go:build !openbsd && !nofuse && !netbsd && !plan9
```
Comment (Jorropo) [inline fuse/node/mount_test.go:1]: Same here

**Review by Jorropo — COMMENTED** [review]
Thx, it's mostly good.

I'll do the changes myself and open a new PR (because of circle CI perms issues it isn't running).

File: coverage/main/main.go (hunk 1 above)
Comment (estensen) [reply coverage/main/main.go]: Someone else beat me to it. Rebased and this went away.

**Comment by Jorropo** [comment]
replacing this PR with #8969 because this one is buggy (it's a usual bug that happen where CI wont run from certain forks)

**Pull request closed without merging**